/*
 * This file was automatically generated by EvoSuite
 * Fri May 05 10:23:47 GMT 2023
 */

package com.example.task_evaluated;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.example.task_evaluated.task_13623;
import java.util.Comparator;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class task_13623_ESTest extends task_13623_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      task_13623 task_13623_0 = new task_13623();
      String[] stringArray0 = new String[1];
      stringArray0[0] = "y/6oloh1t5";
      Object[] objectArray0 = new Object[1];
      objectArray0[0] = (Object) task_13623_0;
      Comparator<Object> comparator0 = (Comparator<Object>) mock(Comparator.class, new ViolatedAssumptionAnswer());
      // Undeclared exception!
      try { 
        task_13623.compareArrays((Object[]) stringArray0, objectArray0, (Comparator<? super Object>) comparator0);
      } catch(ClassCastException e) {
         //
         // Arguments' types prevent them from being compared by this comparator
         //
         verifyException("com.example.task_evaluated.task_13623", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      String[] stringArray0 = new String[4];
      stringArray0[0] = "rS>J2gj(G!#7o mx";
      Comparator<Object> comparator0 = (Comparator<Object>) mock(Comparator.class, new ViolatedAssumptionAnswer());
      doReturn(0, 0, 0, 0).when(comparator0).compare(any() , any());
      int int0 = task_13623.compareArrays(stringArray0, stringArray0, (Comparator<? super String>) comparator0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Integer[] integerArray0 = new Integer[2];
      int int0 = (-1);
      Integer integer0 = new Integer((-1));
      integerArray0[0] = integer0;
      Object[] objectArray0 = new Object[6];
      Comparator<Object> comparator0 = (Comparator<Object>) mock(Comparator.class, new ViolatedAssumptionAnswer());
      doReturn(int0).when(comparator0).compare(any() , any());
      int int1 = task_13623.compareArrays((Object[]) integerArray0, objectArray0, (Comparator<? super Object>) comparator0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Comparator<Object> comparator0 = (Comparator<Object>) mock(Comparator.class, new ViolatedAssumptionAnswer());
      // Undeclared exception!
      try { 
        task_13623.compareArrays((String[]) null, (String[]) null, (Comparator<? super String>) comparator0);
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.example.task_evaluated.task_13623", e);
      }
  }
}
